subgraph/src/fleek-nfa.ts

```diff
@@ -423,23 +426,29 @@ export function handleTransfer(event: TransferEvent): void {
   }
 
   if (parseInt(event.params.from.toHexString()) !== 0) {
-    // Transfer
-
-    // Load the Token by using its TokenId
-    token = Token.load(
-      Bytes.fromByteArray(Bytes.fromBigInt(event.params.tokenId))
-    );
-
-    if (token) {
-      // Entity exists
-      token.owner = owner_address;
-
-      // Save both entities
-      owner.save();
-      token.save();
+    if (parseInt(event.params.to.toHexString()) === 0) {
+      // Burn
+      // Remove the entity from storage
+      // Its controllers and owner will be affected.
+      store.remove('Token', TokenId.toString());
```

Comment:
Do we also need to remove the current owner as an owner?

Reply:
The owner entity can hold multiple access points, tokens, and also be an owner of the collection. When the token is removed, it will also affect the list of the tokens the owner holds since it uses a reverse lookup from the `owner` field of the Token entity.